tf2_ros/src/static_transform_component.cpp

```diff
@@ -0,0 +1,89 @@
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *     * Neither the name of the Willow Garage, Inc. nor the names of its
+ *       contributors may be used to endorse or promote products derived from
+ *       this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "tf2_ros/static_transform_component.hpp"
+#include "rclcpp/rclcpp.hpp"
+
+using namespace std::chrono_literals;
+
+namespace tf2_ros
+{
+StaticTransformPublisher::StaticTransformPublisher(
+  rclcpp::NodeOptions options, const char * name
+)
+: rclcpp::Node(name, options)
+{
+  rclcpp::ParameterValue default_value(static_cast<double>(0));
+  rcl_interfaces::msg::ParameterDescriptor descriptor;
+  descriptor.read_only = true;
+  auto tx = this->declare_parameter("/translation/x", default_value, descriptor);
+  auto ty = this->declare_parameter("/translation/y", default_value, descriptor);
+  auto tz = this->declare_parameter("/translation/z", default_value, descriptor);
+  auto rx = this->declare_parameter("/rotation/x", default_value, descriptor);
+  auto ry = this->declare_parameter("/rotation/y", default_value, descriptor);
+  auto rz = this->declare_parameter("/rotation/z", default_value, descriptor);
+  auto rw = this->declare_parameter("/rotation/w", default_value, descriptor);
+  auto p_frame_id =
+    this->declare_parameter("/frame_id", rclcpp::ParameterValue(std::string("/frame")), descriptor);
+  auto p_child_id =
+    this->declare_parameter("/child_frame_id", rclcpp::ParameterValue(std::string(
+        "/child")), descriptor);
+
+  m_broadcaster = std::make_unique<tf2_ros::StaticTransformBroadcaster>(this);
+  auto clock = std::make_shared<rclcpp::Clock>(RCL_ROS_TIME);
+  rclcpp::TimeSource time_source;
+  time_source.attachClock(clock);
```

Comment:
I don't think this dance is necessary.  Looking at the [code](https://github.com/ros2/rclcpp/blob/2716d3e81ed1bfd8b1afe8d07bdeda7d3a4ea61b/rclcpp/src/rclcpp/node_interfaces/node_clock.cpp#L33), I think the default clock on a node is already an `RCL_ROS_TIME`, so down below you can just call `this->now()`.